src/Spotflow.InMemory.Azure.Storage/Blobs/Internals/BlobClientCore.cs

```diff
@@ -357,7 +424,20 @@ private Task ExecuteAfterHooksAsync<TContext>(TContext context) where TContext :
         return Provider.ExecuteHooksAsync(context);
     }
 
+    private abstract record AcquireBlobError
```

Comment:
All other errors all classes. I do not mind changing them to records but this should be done in different PR for all errors.